lib/internal/worker.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const EventEmitter = require('events');
+const util = require('util');
+
+const { MessagePort, MessageChannel } = process.binding('messaging');
+util.inherits(MessagePort, EventEmitter);
+
+const debug = util.debuglog('worker');
+
+// A MessagePort consists a handle (that wraps around an
+// uv_async_t) which can receive information from other threads and emits
+// .onmessage events, and a function used for sending data to a MessagePort
+// in some other thread.
+function onmessage(payload, flag) {
+  debug(`[${process.threadId}] received message`, flag, payload);
+  // Emit the flag and deserialized object to userland.
+  if (flag === 0 || flag === undefined)
+    this.emit('message', payload);
+  else
+    this.emit('flaggedMessage', flag, payload);
+}
+
+Object.defineProperty(MessagePort.prototype, 'onmessage', {
+  enumerable: false,
+  configurable: true,
+  get() { return onmessage; },
+  set(value) {
+    Object.defineProperty(this, {
+      writable: true,
+      enumerable: true,
+      configurable: true,
+      value
+    });
+    this.ref();
+    this.start();
+  }
+});
```

Comment:
Changing `onmessage` will stop the `message` and `flaggedMessage` events from being emitted. Is that intended behaviour?

Reply:
Yeah, I would say so. This is only there to match the Web spec more closely, but in that case you don’t want the EventEmitter interface anyways I guess.